src/pages/word-duel.jsx

```diff
@@ -5,37 +5,97 @@ import Board from '../components/board';
 
 const MAX_GUESSES = 6;
 
+const EvaluationColor = {
+    Gray: "gray",
+    Yellow: "yellow",
+    Green: "green"
+}
+
 export default function WordDuel() {
     // const [ word, setWord ] = useState(generateNewWord())
-    const word = "awash";
-    const [ guesses, setGuesses ] = useState([]); 
-    const [ error, setError ] = useState("");
+    const word = "stink";
+    const [guesses, setGuesses] = useState([]);
+    const [evaluations, setEvaluations] = useState([]);
+    const [gameOver, setGameOver] = useState(false);
+    const [error, setError] = useState("");
+
     function checkGuess(guess) {
         if (guess.length !== word.length) {
             setError("Guess wrong length!");
             return false;
         }
 
+        if (guesses.length >= MAX_GUESSES) {
+            setError("Guess limit reached.");
+            setGameOver(true);
+            return false;
+        }
+        
         setError("");
-        setGuesses([ ...guesses, guess ])
+        setGuesses([...guesses, guess]);
+        evaluateGuess(guess);
         return true;
     }
 
+    function evaluateGuess(guess) {
+        let evaluation = [];
+        for (let i = 0; i < guess.length; i++) {
```

Comment:
```javascript
const evaluation = guess.split("").map((letter, idx) => {
     if (letter === word[idx]) {
         return EvaluationColor.Green;
     }
     
     if (word.includes(letter)) {
        return EvaluationColor.Yellow;
     }

     return EvaluationColor.Gray;
}, []);
```

Reply:
Added but I have no idea how this works haha